webui/react/src/pages/SignIn.tsx

```diff
@@ -0,0 +1,84 @@
+import { notification } from 'antd';
+import queryString from 'query-string';
+import React, { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import AuthToken from 'components/AuthToken';
+import DeterminedAuth from 'components/DeterminedAuth';
+import Logo, { LogoTypes } from 'components/Logo';
+import Auth from 'contexts/Auth';
+import FullPageSpinner from 'contexts/FullPageSpinner';
+import useAuthCheck from 'hooks/useAuthCheck';
+import usePolling from 'hooks/usePolling';
+import { routeAll } from 'routes';
+import { defaultAppRoute } from 'routes';
+import { locationToPath } from 'utils/routes';
+
+import css from './SignIn.module.scss';
+
+interface Queries {
+  cli?: boolean;
+  redirect?: string;
+}
+
+const SignIn: React.FC = () => {
+  const location = useLocation<{ loginRedirect: Location }>();
+  const auth = Auth.useStateContext();
+  const showSpinner = FullPageSpinner.useStateContext();
+  const setShowSpinner = FullPageSpinner.useActionContext();
+  const queries: Queries = queryString.parse(location.search);
+
+  /*
+   * Check every so often to see if the user is authenticated.
+   * For example, the user can authenticate in a different session,
+   * and this will pick up that auth and automatically redirect them into
+   * their previous app.
+   */
+  const checkAuth = useAuthCheck();
+  const stopPolling = usePolling(checkAuth);
```

Comment:
suggestion: we could probably do this faster than the default polling period as this is a pretty cheap sync check.